### PR TITLE
test: Remove Services SCTP test case

### LINKED ISSUE #16838: CI: K8sServicesTest Checks service across nodes Tests NodePort BPF BPF NAT engine handles unknown protocol packets Should not drop SCTP packets
In https://jenkins.cilium.io/job/Cilium-PR-K8s-1.20-kernel-4.19/918/testReport/junit/Suite-k8s-1/20/K8sServicesTest_Checks_service_across_nodes_Tests_NodePort_BPF_BPF_NAT_engine_handles_unknown_protocol_packets_Should_not_drop_SCTP_packets/:
```
/home/jenkins/workspace/Cilium-PR-K8s-1.20-kernel-4.19/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:518
Expected
    <int>: 0
to be >
    <int>: 0
/home/jenkins/workspace/Cilium-PR-K8s-1.20-kernel-4.19/src/github.com/cilium/cilium/test/k8sT/Services.go:1393
```
The problem here is that iperf transmits no data (0 bytes), which is not greater than 0.

Test output (stderr) shows that iperf3 fails with an error:
```
iperf3: error - control socket has closed unexpectedly
command terminated with exit code 1
```

iperf pod is in hostnetworking mode, if that makes a difference.

Host endpoint has no policy enforcement, but has not seen any traffic either:
```
/sys/fs/bpf/tc/globals/cilium_policy_01097:

POLICY   DIRECTION   IDENTITY   PORT/PROTO   PROXY PORT   BYTES   PACKETS   
Allow    Ingress     0          ANY          NONE         0       0         
Allow    Egress      0          ANY          NONE         0       0         
```

[e3cb6a69_K8sServicesTest_Checks_service_across_nodes_Tests_NodePort_BPF_BPF_NAT_engine_handles_unknown_protocol_packets_Should_not_drop_SCTP_packets (1).zip](https://github.com/cilium/cilium/files/6789208/e3cb6a69_K8sServicesTest_Checks_service_across_nodes_Tests_NodePort_BPF_BPF_NAT_engine_handles_unknown_protocol_packets_Should_not_drop_SCTP_packets.1.zip)

### PR DESCRIPTION
**Pull request by brb**
The test case is supposed to test whether the BPF NAT engine doesn't
drop packets of a type which it doesn't support (e.g. SCTP). However,
recently the test case became flaky. Considering this and a little value
the test provides, let's remove the test case.

Instead, we should create a BPF unit test for it once it's available
(#16862).

I've reopened https://github.com/cilium/cilium/issues/10541 and updated the description.

Fix #16838
Fix #16839

### CODE REVIEW

**Comment by brb** [comment]
test-1.16-netnext

**Comment by brb** [comment]
net-next hit https://github.com/cilium/cilium/issues/12511. Considering that the test removal doesn't have anything to do with it, no need to re-run tests.

File: test/k8sT/Services.go [hunk 1]
```diff
@@ -1357,45 +1356,6 @@ Secondary Interface %s :: IPv4: (%s, %s), IPv6: (%s, %s)`, helpers.DualStackSupp
...
-				// GKE COS image does not support sctp module.
-				SkipContextIf(func() bool {
-					return helpers.DoesNotRunWithKubeProxyReplacement() || helpers.RunsOnGKE()
-				}, "BPF NAT engine handles unknown protocol packets", func() {
-					var (
-						iperf3Manifest string
-					)
-					BeforeAll(func() {
-						DeployCiliumAndDNS(kubectl, ciliumFilename)
-						iperf3Manifest = helpers.ManifestGet(kubectl.BasePath(), "iperf3-deployment.yaml")
```
Comment (tklauser) [inline test/k8sT/Services.go:1369]: I think we can remove the `iperf3-deployment.yaml` manifest as well. This seems to be the only test using it.

**Review by tklauser — APPROVED** [review]
LGTM apart from a comment about `iperf3-deployment.yaml`.

**Review by kkourt — APPROVED** [review]

File: test/k8sT/Services.go (hunk 1 above)
Comment (brb) [reply test/k8sT/Services.go:1369]: Oops, good catch! Removed.

**Comment by brb** [comment]
Considering that the new change after the net-next had passed is the removal of the manifest file (not used by any of the tests), marking it as ready to merge.

**Review by tklauser — APPROVED** [review]

**Pull request merged**